### PR TITLE
[proof] Deprecate unused tacmach functions.

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Review by ppedrot — COMMENTED** [review]
I don't think the comment is right, it should rather read something like "use the New variants".

**Comment by ejgallego** [comment]
Ouch, I thought this was fixed. Still a couple of cases without replacement tho.

File: proofs/tacmach.mli [hunk 1]
```diff
@@ -49,22 +50,33 @@ val pf_eapply : (env -> evar_map -> 'a -> evar_map * 'b) ->
...
 val pf_whd_all       : Goal.goal sigma -> constr -> constr
+[@@ocaml.deprecated "Use the version in Tacmach.New"]
 val pf_hnf_constr              : Goal.goal sigma -> constr -> constr
+[@@ocaml.deprecated "Use the version in Tacmach.New"]
 val pf_nf                      : Goal.goal sigma -> constr -> constr
+[@@ocaml.deprecated "Use the version in Tacmach.New"]
 val pf_nf_betaiota             : Goal.goal sigma -> constr -> constr
 val pf_reduce_to_quantified_ind : Goal.goal sigma -> types -> (inductive * EInstance.t) * types
 val pf_reduce_to_atomic_ind     : Goal.goal sigma -> types -> (inductive * EInstance.t) * types
+[@@ocaml.deprecated "what's the replacement?"]
```
Comment (ppedrot) [inline proofs/tacmach.mli]: Tacred.reduce_to_atomic_ind

File: proofs/tacmach.mli [hunk 2]
```diff
@@ -49,22 +50,33 @@ val pf_eapply : (env -> evar_map -> 'a -> evar_map * 'b) ->
...
 val pf_compute                 : Goal.goal sigma -> constr -> constr
+[@@ocaml.deprecated "Use the version in Tacmach.New"]
 val pf_unfoldn    : (occurrences * evaluable_global_reference) list
   -> Goal.goal sigma -> constr -> constr
+[@@ocaml.deprecated "What's the replacement?"]
```
Comment (ppedrot) [inline proofs/tacmach.mli]: Tacred.unfoldn

File: proofs/tacmach.mli [hunk 3]
```diff
@@ -49,22 +50,33 @@ val pf_eapply : (env -> evar_map -> 'a -> evar_map * 'b) ->
...
 
 val pf_const_value : Goal.goal sigma -> pconstant -> constr
+[@@ocaml.deprecated "What's the replacement?"]
```
Comment (ppedrot) [inline proofs/tacmach.mli]: Environ.constant_value_in

**Comment by ppedrot** [comment]
In general, even the `New.pf_*` variants are dubious. We provide a goal structure in the engine, but it's mainly for backwards compatibility. This API is error-prone w.r.t. evar-maps, so I'd even go as far as removing the `New.pf_*` variants at some point.

**Comment by ejgallego** [comment]
Deprecation notices added, thanks for the hints!

> In general, even the `New.pf_*` variants are dubious. We provide a goal structure in the engine, but it's mainly for backwards compatibility. This API is error-prone w.r.t. evar-maps, so I'd even go as far as removing the `New.pf_*` variants at some point.

Indeed, that would be great, but seems like a daunting task. Tho, pushing tactics upwards actually would make this easier, or at least to have a `Tactics2` library without them.

**Review by ppedrot — APPROVED** [review]

**Pull request merged**